sdcm/tester.py

```diff
@@ -1519,7 +1519,11 @@ def run_stress_cassandra_thread(
             self, stress_cmd, duration=None, stress_num=1, keyspace_num=1, profile=None, prefix='', round_robin=False,
             stats_aggregate_cmds=True, keyspace_name=None, stop_test_on_failure=True, **_):
         # stress_cmd = self._cs_add_node_flag(stress_cmd)
-        timeout = self.get_duration(duration)
+        if duration:
+            timeout = self.get_duration(duration)
+        else:
+            timeout = get_timeout_from_stress_cmd(stress_cmd) or self.get_duration(duration)
```

Comment:
this might be a problem, because in case we do write without `duration`, but expect to write a specific number of keys, in big data set tests, it can take hours, so the 10 min we give here will kill these tests

Reply:
If this is a problem, should we have bumped into it on master?

Reply:
it's default to test duration which is long enough.

but if it's really stuck, test duration is also too big, since in rolling upgrade to are execute in the midst of the test.

Reply:
@fruch It might be even too long for longevities. We use `test_duration` as the `timeout` for getting the stress futures, for the remoter cmd and for the results verification (all in the `CassandraStressThread` class). If any of these get's stuck or is slower, we're guaranteed to end the test with a critical test timeout. Or maybe I'm missing something here?